pyiceberg/io/pyarrow.py

```diff
@@ -1772,16 +1772,17 @@ def write_file(io: FileIO, table_metadata: TableMetadata, tasks: Iterator[WriteT
     )
 
     def write_parquet(task: WriteTask) -> DataFile:
+        arrow_table = pa.Table.from_batches(task.record_batches)
+        df = to_requested_schema(table=arrow_table, from_schema=iceberg_table_schema, to_schema=parquet_schema)
```

Comment:
similar to the read side (#83 & #597), turns the Arrow table from unsanitized Iceberg table schema to sanitized parquet schema

Reply:
Currently, we batch the incoming dataframe first (in `_dataframe_to_data_files`) and then transform the scheme for each batch. 

We can optimize by transforming first and then batching. 

I want the schema transformation to happen as closely to the parquet writing as possible, so going with the first method for now